lib/src/marker_layer.dart

```diff
@@ -64,32 +63,28 @@ class _MarkerLayerState extends State<MarkerLayer>
         children: (List<Marker> markers) sync* {
           for (final m in markers) {
             // Resolve real alignment
-            final left =
-                0.5 * m.width * ((m.alignment ?? Alignment.center).x + 1);
-            final top =
-                0.5 * m.height * ((m.alignment ?? Alignment.center).y + 1);
+            final left = 0.5 * m.width * ((m.alignment ?? Alignment.center).x + 1);
+            final top = 0.5 * m.height * ((m.alignment ?? Alignment.center).y + 1);
             final right = m.width - left;
             final bottom = m.height - top;
 
             // Perform projection
-            final pxPoint = map.project(m.point);
+            final pxPoint = map.projectAtZoom(m.point);
+
+            final width = (pxPoint.dx + left) - (pxPoint.dx - right);
+            final height = (pxPoint.dy + top) - (pxPoint.dy - bottom);
 
             // Cull if out of bounds
-            if (!map.pixelBounds.containsPartialBounds(
-              Bounds(
-                Point(pxPoint.x + left, pxPoint.y - bottom),
-                Point(pxPoint.x - right, pxPoint.y + top),
-              ),
-            )) continue;
+            if (!map.pixelBounds.contains(Offset(width, height))) {
+              continue;
```

Comment:
Marker always out of bounds ((
After calculation in rows 74,75 **_width_** always equals **_m.width_** and **_height_** always equals **_m.height_** 
That corresponds just size of marker, but not a position